site/content/entry/yuzu-progress-report-feb-2023/index.md

```diff
@@ -0,0 +1,381 @@
+But as a temporary solution, this change makes playing affected games much more tolerable.
+Work continues!
+
+## CPU, kernel, services, and optimizations
+
+[Merry](https://github.com/merryhime) {{< gh-hovercard "9735" "updated dynarmic" >}}, our JIT, to version 6.4.5, solving some issues and improving compilation times.
+One game that benefited from this is `Taiko Risshiden V DX`, which got its load times reduced considerably.
+
+{{< single-title-imgs
+    "Take your time (Taiko Risshiden V DX)"
+    "./taiko1.png"
+    "./taiko2.png"
+    >}}
+
+byte[] has also been going deep into the kernel code. One of his changes improves the `SVC wrappers`.
+But what are the wrappers? Well, games and homebrew need to talk to the kernel, and to do so they use SVC instructions, which act as an interface with the kernel.
+
+{{< imgs
+	"./wrapper.png| The return of mspaint"
+  >}}
+
+The work of the wrapper is to translate a request from the game into a C++ call into the kernel, and then get the result from the kernel back into the game.
+In the past yuzu used manually written wrappers that were very error-prone. With this change, byte[] instead {{< gh-hovercard "9742" "automatically generates" >}} the needed wrappers.
+This code is far more accurate, so if you find games that no longer crash or soft-lock, here’s the new code responsible for it.
+One such example is `Moero Crystal H` which starts booting for the first time.
+
+{{< imgs
+	"./mch.png| Don’t judge, it’s part of the Switch’s library (Moero Crystal H)"
+  >}}
+
+This game requires more work to get proper video and gameplay rendering.
+
+Keeping the pursuit for accuracy, byte[] also implemented {{< gh-hovercard "9832" "HLE multiprocess" >}} for services.
+Now instead of pushing the game’s requests to the services, the services wait for the requests to arrive.
+All this processing is done, for the most part, in what would be the emulated `core 3` of the Switch, instead of all over the place with one thread per service. 
+Games usually only have access to cores 0 to 2, so we're now dedicating the last core for system service processing, just as the actual Switch operating system does!
+
+This part of CPU emulation is one of the main reasons we recommend at least 6 cores in our [hardware requirements](https://yuzu-emu.org/help/quickstart/#hardware-requirements), 4 for uninterrupted emulation of the Switch’s CPU, and extras for other processes and tasks. Just a 4 core CPU will be usually overburdened. HT/SMT may help, but that will always depend on the workload at any given moment.
+A SMT/HT thread can’t improve performance in a significant way if the core is already saturated.
+
+Boss [bunnei](https://github.com/bunnei) chimed in too, fixing a mistake in {{< gh-hovercard "9773" "memory mapping." >}}
+We erroneously used the (bear with me, it’s not a redundancy) system’s system resource tracking, instead of relying on the application’s one when loading programs into memory.
```

Comment:
```suggestion
We erroneously used the (bear with me, it’s not a redundancy) system’s system resource tracking, instead of relying on the application’s resource tracking when loading programs into memory.
```